AudioStreaming/Streaming/AudioPlayer/AudioPlayer.swift

```diff
@@ -150,12 +151,14 @@ open class AudioPlayer {
                                                        rendererContext: rendererContext,
                                                        outputAudioFormat: outputAudioFormat.basicStreamDescription)
 
-        frameFilterProcessor = FrameFilterProcessor(mixerNode: audioEngine.mainMixerNode)
-
         playerRenderProcessor = AudioPlayerRenderProcessor(playerContext: playerContext,
                                                            rendererContext: rendererContext,
                                                            outputAudioFormat: outputAudioFormat.basicStreamDescription)
 
+        
+        frameFilterProcessor = FrameFilterProcessor(mixerNodeProvider: {
+            engine.mainMixerNode
+        })
```

Comment:
It could also be possible that the frameFilterProcessor be initialised when a relevant method is first used. But we could improve on another PR.